modelexpress_client/python/modelexpress/engines/vllm/artifacts.py

```
@@ -0,0 +1,571 @@
+            max_mtime_ns = max(max_mtime_ns, stat.st_mtime_ns)
+    except OSError:
+        return None
+    if count == 0:
+        return None
+    return count, total_size, max_mtime_ns
+
+
+def _vllm_health_ready() -> bool:
+    url = _vllm_health_url()
+    try:
+        with urllib.request.urlopen(url, timeout=1.0) as response:
```

Ruff (0.15.18)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)